torch_geometric/graphgym/train.py

```diff
@@ -69,13 +70,14 @@ def train(loggers, loaders, model, optimizer, scheduler):
     split_names = ['val', 'test']
     for cur_epoch in range(start_epoch, cfg.optim.max_epoch):
         train_epoch(loggers[0], loaders[0], model, optimizer, scheduler)
-        loggers[0].write_epoch(cur_epoch)
+        if is_train_eval_epoch(cur_epoch):
```

Comment:
Thanks for your reply! I've been using GraphGym to evaluate some large datasets (node-level tasks) including the OGBN ``arxiv`` and ``proteins``. Due to a large number of nodes in these graphs, the computation of auroc is very expensive and I've observed a severe GPU usage bottleneck (near 0% usage) because of this. Skipping the training epoch evaluation could effectively remedy this issue. Furthermore, I don't see the evaluations for the training epoch being used anywhere in the downstream analysis, so I thought it might not hurt to disable it in favor of improving the GPU usage.

It has also been suggested to look into the [torchmetric](https://torchmetrics.readthedocs.io/en/latest/) in https://github.com/pyg-team/pytorch_geometric/issues/3349#issuecomment-946354517, the metric refactoring in this PR should also enable this in a future PR.

Reply:
Thanks for explaining! This makes sense. And indeed, future PRs on `torchmetric` is welcomed!